shops/code/controllers/shopController.js

```diff
@@ -0,0 +1,137 @@
+import { JSONFilePreset } from "lowdb/node";
+
+// Read or create db.json
+// defaultData specifies the structure of the database
+const defaultData = { meta: {"tile": "List of shops","date": "November 2024"}, shops : [] }
+const db = await JSONFilePreset('db.json', defaultData)
+const shops = db.data.shops;
+
+export async function getAvailableId() {
+  const ids = shops.map(shop => shop.id);
+  const availableId = Math.max(...ids) + 1;
+  return availableId;
+}
+
+export async function createShop(req, res) {
+  const id = getAvailableId();
+  const address = {
+    city: req.body.location.city,
+    address: req.body.location.address
+  }
+  const name = req.body.name;
+  const phoneNumber = req.body.phonenumber;
+  const openingHours = {
+    monday: req.body.openingHours.monday,
+    tuesday: req.body.openingHours.tuesday,
+    wednesday: req.body.openingHours.wednesday,
+    thursday: req.body.openingHours.thursday,
+    friday: req.body.openingHours.friday,
+    saturday: req.body.openingHours.saturday,
+    sunday: req.body.openingHours.sunday
+  };
+  const payingMethods = req.body.payingMethods;
+  const userID = req.body.userID;
+
+  if (!name || !address.city || !address.address || !userID || !openingHours) {
+    return res.status(400).send('Missing required fields');
+  } else {
+    shops.push({
+      id: id,
+      name: name,
+      address: address,
+      phoneNumber: phoneNumber,
+      openingHours: openingHours,
+      payingMethods: payingMethods,
+      userID: userID
+    });
+    await db.write();
+
+    res.status(201).send(`Shop created with name: ${name} `);
+  }
+}
+/**
+ * aquire list of shops
+ * @param {*} req 
+ * @param {*} res 
+ */
+export async function responseShops(req, res) {
+  res.status(200).send(shops);
+}
+
+/**
+ * this function returns a shop by id or name
+ * @param {*} param 
+ */
+export async function responseShop(req, res) {
+  const param = req.params.param;
+  const shop = shops.find(shop => shop.id === Number(param) || shop.name === param);
+
+  if (shop) {
+    res.status(200).send(shop);
+  } else {
+    res.status(404).send('Shop not found');
+  }
+}
+
+
+export async function updateShop(req, res) {
+  const id = parseInt(req.params.id); // Extract the shop ID from the URL parameter
+  const shop = markets.find(shop => shop.id === id); // Find the shop by ID
+
+  if (!shop) {
+    return res.status(404).send(`Shop with ID: ${id} not found`);
+  }
+
+  // Update shop attributes based on the request body
+  if (req.body.name) {
+    shop.name = req.body.name;
+  }
+
+  if (req.body.location) {
+    shop.address = {
+      city: req.body.location.city || shop.address.city,
+      address: req.body.location.address || shop.address.address
+    };
+  }
+
+  if (req.body.phonenumber) {
+    shop.phoneNumber = req.body.phonenumber;
+  }
+
+  if (req.body.openingHours) {
```

Comment:
de code werkt maar als je maandag bijvoorbeeld express leeg laat en de rest niet blijft maandag op de oude waarden staan dus bijvoorbeeld. oude waardes ( ma : 10u di 11 wo 11 do 11 vrij11) nieuwe waardes (ma laat je leeg omdat je bijvoorbeeld gesloten bent  di11 wo11 do11 vrij 11) dan krijg je als output nu ma 10 di 11 wo 11 do 11 vrij 11 terwijl je wilt dat maandag leeg is omdat je dicht bent dus je code werkt je moet alleen in je achterhoofd halen dat dit kan gebeuren dus moet je nadenken over hoe je mensen laat aangeven dat ze gesloten zijn en anders kan je ook kijken naar spread operators